src/t/d4.clj

```diff
@@ -0,0 +1,164 @@
+(ns t.d4
+  (:gen-class)
+  (:require
+    [clojure.string :as cljstr]
+    [clojure.set :as cljset]))
+
+(def sample
+"MMMSXXMASM
+MSAMXMSMSA
+AMXSXMAAMM
+MSAMASMSMX
+XMASAMXAMM
+XXAMMXXAMA
+SMSMSASXSS
+SAXAMASAAA
+MAMMMXMMMM
+MXMXAXMASX")
+
+(defn parse-input
+  [input]
+  (->> input
+       cljstr/split-lines
+       (map-indexed (fn [i l]
+                      (->> l
+                      (map-indexed (fn [ii letter] {letter [[i ii]]}))
```

Comment:
It's a minor thing but I found `i` and `ii` a bit hard to parse. Perhaps:
```suggestion
       (map-indexed (fn [y l]
                      (->> l
                      (map-indexed (fn [x letter] {letter [[y x]]}))
```
or `line` & `col`, but it's a bit awkward here with `lines` and `l `.

Reply:
I understand.
I won't change it though.
This i, ii, ... comes from a ... tweet or blog or something wondering why are we using i, then j, than k, and so on in loops, when you could have the level of your loop in a ore obvious way.

Reply:
That's actually a nice trick. 🤔